src/awx_plugins/credentials/github_app.py

```diff
@@ -0,0 +1,190 @@
+"""GitHub App Credential Plugin.
+
+This module defines a credential plugin for GitHub Apps, allowing
+authentication via GitHub App tokens.
+
+Functions:
+
+- ``github_app_backend``: Generates a GitHub App token fields from GitHub Admin.
+- ``github_app_lookup``: Defines the credential plugin interface.
+"""
+
+from typing import TypedDict
+
+from awx_plugins.interfaces._temporary_private_api import (  # noqa: WPS436
+    ManagedCredentialType,
+)
+from awx_plugins.interfaces._temporary_private_django_api import (  # noqa: WPS436
+    gettext_noop as _,
+)
+
+from github import Auth, Github
+
+from .plugin import CredentialPlugin
+
+
+class FieldDict(TypedDict, total=False):
+    id: str
+    label: str
+    type: str
+    format: str  # Optional field
+    secret: bool  # Optional field
+    multiline: bool  # Optional field
```

Comment:
> `total=False`

TIL!

Though, apparently, there's a way to only mark some filed as optional:
```suggestion
class FieldDict(TypedDict):
    id: str
    label: str
    type: str
    format: NotRequired[str]
    secret: NotRequired[bool]
    multiline: NotRequired[bool]
```
*(https://typing.readthedocs.io/en/latest/spec/typeddict.html#required-notrequired)*